packages/node/src/types.ts

```diff
@@ -209,14 +212,60 @@ export interface WorkerResponse {
 // ===========================================
 // Events
 // ===========================================
-export interface EVMEventLogWithMetadata {
+interface EVMEventLogMetadata {
   readonly blockNumber: number;
   readonly currentBlock: number;
   readonly ignoreBlockedRequestsAfterBlocks: number;
-  readonly parsedLog: ethers.utils.LogDescription;
   readonly transactionHash: string;
 }
 
+// Maybe there will be less hacky way to obtain this in the future.
+// See: https://github.com/ethereum-ts/TypeChain/issues/376
+// NOTE: I am also ignoring the typed tupple and only extracting the typed event object.
+type ExtractTypedEvent<T> = T extends TypedEventFilter<any, infer EventArgsObject> ? EventArgsObject : never;
+export type AirnodeRrpEvents = Pick<
+  InstanceType<typeof AirnodeRrp>['filters'],
+  | 'ClientRequestCreated'
+  | 'ClientFullRequestCreated'
+  | 'ClientRequestFulfilled'
+  | 'ClientRequestFailed'
+  | 'WithdrawalRequested'
+  | 'WithdrawalFulfilled'
+>;
+export type AirnodeRrpLog<T extends keyof AirnodeRrpEvents> = ExtractTypedEvent<ReturnType<AirnodeRrpEvents[T]>>;
+
+export type AirnodeLogDescription<T> = Omit<ethers.utils.LogDescription, 'args'> & { args: T };
+
+export interface EVMFullApiRequestCreatedLog extends EVMEventLogMetadata {
+  parsedLog: AirnodeLogDescription<AirnodeRrpLog<'ClientFullRequestCreated'>>;
+}
+
+export interface EVMTemplateRequestCreatedLog extends EVMEventLogMetadata {
+  parsedLog: AirnodeLogDescription<AirnodeRrpLog<'ClientRequestCreated'>>;
+}
+
+export type EVMRequestCreatedLog = EVMTemplateRequestCreatedLog | EVMFullApiRequestCreatedLog;
+
+export interface EVMRequestFulfilledLog extends EVMEventLogMetadata {
+  parsedLog:
+    | AirnodeLogDescription<AirnodeRrpLog<'ClientRequestFulfilled'>>
+    | AirnodeLogDescription<AirnodeRrpLog<'ClientRequestFailed'>>;
+}
+
+export interface EVMWithdrawalRequestLog extends EVMEventLogMetadata {
+  parsedLog: AirnodeLogDescription<AirnodeRrpLog<'WithdrawalRequested'>>;
+}
+
+export interface EVMWithdrawalFulfilledLog extends EVMEventLogMetadata {
+  parsedLog: AirnodeLogDescription<AirnodeRrpLog<'WithdrawalFulfilled'>>;
+}
+
+export type EVMEventLog =
```

Comment:
The idea is to have the events which are used in node package typed. This is useful for many things - autocomplete, refactorings, static guarantees imposed by TS